eng/common/scripts/modules/Package-Properties.psm1

```diff
@@ -0,0 +1,243 @@
+        {
+            $this.pkgReadMePath = Join-Path $pkgDirectoryPath "README.md"
+        } else
+        {
+            $this.pkgReadMePath = $null
+        }
+
+        if (Test-Path (Join-Path $pkgDirectoryPath "CHANGELOG.md"))
+        {
+            $this.pkgChangeLogPath = Join-Path $pkgDirectoryPath "CHANGELOG.md"
+        } else
+        {
+            $this.pkgChangeLogPath = $null
+        }
+    }
+}
+
+Install-Module -Name powershell-yaml -RequiredVersion 0.4.1 -Force -Scope CurrentUser
+
+function Extract-PkgProps ($pkgPath, $pkgName)
+{
+    if ($Language -eq "net")
+    { Extract-DotNetPkgProps -pkgPath $pkgPath -pkgName $pkgName 
+    }
+    if ($Language -eq "java")
+    { Extract-JavaPkgProps -pkgPath $pkgPath -pkgName $pkgName 
+    }
+    if ($Language -eq "js")
+    { Extract-JsPkgProps -pkgPath $pkgPath -pkgName $pkgName 
+    }
+    if ($Language -eq "python")
+    { Extract-PythonPkgProps -pkgPath $pkgPath -pkgName $pkgName 
+    }
+}
+
+function Extract-DotNetPkgProps ($pkgPath, $pkgName)
+{
+    $projectPath = Join-Path $pkgPath "src" "$pkgName.csproj"
+    if (Test-Path $projectPath)
+    {
+        $projectData = New-Object -TypeName XML
+        $projectData.load($projectPath)
+
+        $pkgVersion = Select-XML -Xml $projectData -XPath '/Project/PropertyGroup/Version'
+        
+        return [PackageProps]::new($pkgName, $pkgVersion, $pkgPath)
+    } else 
+    {
+        return $null
+    }
+}
+
+function Extract-JsPkgProps ($pkgPath, $pkgName)
+{
+    $projectPath = Join-Path $pkgPath "package.json"
+    if (Test-Path $projectPath)
+    {
+        $projectJson = Get-Content $projectPath | Out-String | ConvertFrom-Json
+        $jsStylePkgName = $pkgName.replace("azure-", "@azure/")
+        if ($projectJson.name -eq "$jsStylePkgName")
+        {
+            return [PackageProps]::new($projectJson.name, $projectJson.version, $pkgPath)
+        }
+    }
+    return $null
+}
+
+function Extract-PythonPkgProps ($pkgPath, $pkgName)
+{
+    $pkgName = $pkgName.Replace('_', '-')
+
+    if (Test-Path (Join-Path $pkgPath "setup.py"))
+    {
+        $setupLocation = $pkgPath.Replace('\','/')
+        pushd $RepoRoot
+        $setupProps = (python -c "import scripts.devops_tasks.common_tasks; obj=scripts.devops_tasks.common_tasks.parse_setup('$setupLocation'); print('{0},{1}'.format(obj[0], obj[1]));") -split ","
+        Write-Host $setupProps
+        popd
+        if($setupProps.Length -ne 2) 
+        {
+            Write-Error ("Error parsing {0}" -f (Join-Path $pkgPath "setup.py"))
+            exit 1
+        }
+        if (($setupProps -ne $null) -and ($setupProps[0] -eq $pkgName))
+        {
+            return [PackageProps]::new($setupProps[0], $setupProps[1], $pkgPath)
+        }
+    }
+    return $null
+}
+
+function Extract-JavaPkgProps ($pkgPath, $pkgName)
+{
+    $projectPath = Join-Path $pkgPath "pom.xml"
+
+    if (Test-Path $projectPath)
+    {
+        $projectData = New-Object -TypeName XML
```

Comment:
@chidozieononiwu, I think you can do this in a slightly easier way similar to what I did for the [doc index generation](https://github.com/Azure/azure-sdk-for-java/blob/master/eng/docgeneration/Generate-DocIndex.ps1#L73). The Select-XML statements aren't necessary, once you've created the XML object and called load the on the pom file, you should only need to access $projectData.project.artifactId and $projectData.project.version to get the artifact and version entries. Also, the values returned from accessing them this way are string  and the ToString() conversions would not be necessary.

Reply:
Thanks